static/js/signal.js

```diff
@@ -0,0 +1,67 @@
+updateSignal();
+
+const refreshTime =2 * 1000
+const clearSignalAttempts = 3 
+const maxStrength = 100;
+const minStrength = 0;
+const bars = 5;
+const signalRange = maxStrength - minStrength
+const signalPerBar = signalRange / parseFloat(bars);
+
+var signalFails = 0;
+
+async function updateSignal() {
+  var xmlHttp = new XMLHttpRequest();
+  xmlHttp.open("GET", "/api/signal-strength", true);
+  xmlHttp.setRequestHeader("Authorization", "Basic "+btoa("admin:feathers"))
+  
+  xmlHttp.onload = async function() {
+    if (xmlHttp.status == 200) {
+      handleSignalSuccess(xmlHttp.response);
+    } else {
+      handleSignalFailure("status:" + xmlHttp.status + " response:" + xmlHttp.response);
+    }
+    reCheckSignal(refreshTime); 
+  }
+
+  xmlHttp.onerror = function() {
+    handleFailure("error occured accesing " + "/api/signal-strength")
+    reCheckSignal(refreshTime); 
+  }
+
+  xmlHttp.send(null);
+}
+
+function handleSignalSuccess(signalVal){
+  signalFails = 0;
+  var signalElement = document.getElementById("signal-strength");
+  var strength =parseInt(signalVal);
+
+  signalElement.innerText = strength;
```

Comment:
The 0 - 5 value is just an abstract value. I don't think it's worth displaying by itself. Just use it to decide which icon to display.

With that in mind, you can probably simplfiy the consts at the top of this file and the calculations below.